### PR TITLE
[ZEPPELIN-5376] Polish server and service package codes of zeppelin-server

### PR DESCRIPTION
**Pull request by cuspymd**
### What is this PR for?
Polish server and service package codes of zeppelin-server
- Delete unused import
- Delete and optimize redundant codes
- Add `final` keyword to some variables that have unchanged reference.

### What type of PR is it?
[Refactoring]

### What is the Jira issue?
* [ZEPPELIN-5376] Polish server and service package codes of zeppelin-server

### How should this be tested?
* CI

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 import org.slf4j.LoggerFactory;
 
 public class NoAuthenticationService implements AuthenticationService {
-  private static Logger logger = LoggerFactory.getLogger(NoAuthenticationService.class);
+  private static final Logger logger = LoggerFactory.getLogger(NoAuthenticationService.class);
```
Comment (zjffdu) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java:33]: Rename it to be `LOGGER`, this is our convention for static final variables.

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java [hunk 2]
```diff
@@ -407,8 +407,8 @@ public boolean isAuthenticated() {
     Connection con = null;
     PreparedStatement ps = null;
     ResultSet rs = null;
-    DataSource dataSource = null;
-    String authQuery = "";
+    DataSource dataSource;
+    String authQuery;
```
Comment (zjffdu) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java:411]: Only change these 2 variables ?

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java (hunk 1 above)
Comment (cuspymd) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java:33]: In the example in the [Constant names](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names) section of the Google java style document, it says that `Logger` is not a constant. Could you check it out again?
> // Not constants
...
static final Logger logger = Logger.getLogger(MyClass.getName());

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java (hunk 2 above)
Comment (cuspymd) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java:411]: In the case of `con`, `ps`, and `rs` variables, they are used in the `finally` block, so if not initialized, a compilation error occurs.

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java (hunk 1 above)
Comment (zjffdu) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java:33]: I think the logger in google java style document refer to `java.util.logging.Logger` which is a class and do has method to change its state (e.g. Logger#config). But what zeppelin use is `org.slf4j.Logger` is an interface which has no method to change its state.

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/JobManagerService.java [hunk 3]
```diff
@@ -65,9 +65,10 @@ public JobManagerService(Notebook notebook,
     Note jobNote = notebook.getNote(noteId);
     if (jobNote == null) {
       callback.onFailure(new IOException("Note " + noteId + " not found"), context);
+    } else {
+      notesJobInfo.add(new NoteJobInfo(jobNote));
+      callback.onSuccess(notesJobInfo, context);
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/JobManagerService.java:70]: This is a change of programme. But it looks good.

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java (hunk 1 above)
Comment (cuspymd) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NoAuthenticationService.java]: I changed it.

**Comment by zjffdu** [comment]
LGTM

**Pull request closed without merging**